src/ai/util/areaCodeFromZip.util.ts

```diff
@@ -0,0 +1,230 @@
+import { Injectable, Logger } from '@nestjs/common'
+import { differenceInYears, formatISO } from 'date-fns'
+import { S3Service } from 'src/vendors/aws/services/s3.service'
+import { z } from 'zod'
+import { AiService } from '../ai.service'
+
+const ZIP_TO_AREA_CODE_FILE = 'zip-to-area-code-mappings.json'
+const CACHE_EXPIRY_YEARS = 1
+
+const AreaCodeResponseSchema = z.array(
+  z.string().regex(/^\d{3}$/, 'Area code must be a 3-digit number'),
+)
+
+type ZipToAreaCodeEntry = {
+  areaCodes: string[]
+  lastFetchedAt: string // ISO datestring
+}
+
+type ZipToAreaCodeMapping = Record<string, ZipToAreaCodeEntry>
+
+@Injectable()
+export class AreaCodeFromZipService {
+  private readonly logger = new Logger(AreaCodeFromZipService.name)
+  private readonly zipToAreaCodeBucket: string
+
+  constructor(
+    private readonly s3Service: S3Service,
+    private readonly aiService: AiService,
+  ) {
+    const ZIP_TO_AREA_CODE_BUCKET = process.env.ZIP_TO_AREA_CODE_BUCKET as string
+    if (!ZIP_TO_AREA_CODE_BUCKET) {
+      throw new Error('ZIP_TO_AREA_CODE_BUCKET environment variable is required')
+    }
+    this.zipToAreaCodeBucket = ZIP_TO_AREA_CODE_BUCKET
+  }
```

Comment:
RE: https://github.com/thegoodparty/gp-api/pull/860#discussion_r2543368035

Looks like this just got moved up into the constructor.. 

This should be at the top of the file, completely outside of the class to match the established pattern, the likes of which can be seen in this example:
https://github.com/thegoodparty/gp-api/blob/develop/src/ai/ai.service.ts#L17-L26